### ISSUE TITLE
Yarn 1.9 ignores .npmrc registry on scoped packages with `yarn create` and `yarn global add`

### ISSUE THREAD

**Issue by astorije** [yarnpkg/yarn#6212]
**Do you want to request a *feature* or report a *bug*?**

Bug

**What is the current behavior?**

With `yarn 1.9.2`, using the following `~/.npmrc` info, which informs `yarn` that `@foo` scoped packages must be fetched from a private registry:

```
$ grep @foo ~/.npmrc
@foo:registry = "https://artifactory.foobar.com/"
```

`yarn create @foo/bar` and `yarn global add @foo/bar` try to fetch from the default registry instead:

```
/tmp/foobar $ yarn -v
1.9.2

/tmp/foobar $ yarn create @foo/bar
yarn create v1.9.2
[1/4] 🔍  Resolving packages...
error An unexpected error occurred: "https://registry.yarnpkg.com/@foo%2fcreate-bar: Not found".
```

That exact same process was running fine under `yarn` v1.7.0:

```
/tmp/foobar $ yarn -v
1.7.0

/tmp/foobar $ yarn create @foo/bar
yarn create v1.7.0
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 📃  Building fresh packages...
success Installed "@foo/create-bar@0.5.0" with binaries:
      - create-bar
[...]
```

These were run on the same machine, and the only thing that was run between the successful 1.7.0 run and failing 1.9.2 run was upgrading `yarn` using `brew`.

Note that `yarn add @foo/create-bar` (i.e. locally) fetched the package from the correct registry.

**If the current behavior is a bug, please provide the steps to reproduce.**

Reproducing this involves a bit of setup, since for us it manifests with a private registry and scoped packages. Hopefully the details above provide enough information.

**What is the expected behavior?**

Expected behavior is that the package gets fetched from the correct registry when it is scoped with `@foo`.

**Please mention your node.js, yarn and operating system version.**

This was confirmed on both Node v8 and Node v10, tested under `yarn` v1.7.0 and v1.9.2, all on macOS High Sierra.

**Comment by astorije** [comment]
Only recent change I can see to `yarn create` is https://github.com/yarnpkg/yarn/pull/5983 and incidentally it affects scoped packages. Does this ring a bell to you, @chriskrycho?

**Comment by arcanis** [comment]
Thanks for the report!

Duplicate of #6174, fixed by #6207 - it will be in the 1.9.3, which I'm in the process of releasing at the moment. I'll close this issue to avoid having to monitor multiple ones, please suscribe on #6174 🙂

**Comment by astorije** [comment]
Oh, perfect! Thanks @arcanis and sorry for the duplicate, I searched through issues before opening this ticket and missed this one, my bad!
